### PR TITLE
update template.yaml to include sqs event changes

### PR DESCRIPTION
**Pull request by MarieseMikely**
### What changed

Created mock sqs queue for the dev environment 

### Why did it change

To test the `Headers` value in the event messages for Address, KBV and Check CRIs 

### Issue tracking

- [OJ-2545](https://govukverify.atlassian.net/browse/OJ-2545)
- [OJ-2546](https://govukverify.atlassian.net/browse/OJ-2546)
- [OJ-2547](https://govukverify.atlassian.net/browse/OJ-2547)

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2545]: https://govukverify.atlassian.net/browse/OJ-2545?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OJ-2546]: https://govukverify.atlassian.net/browse/OJ-2546?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OJ-2547]: https://govukverify.atlassian.net/browse/OJ-2547?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/lambda/template.yaml [hunk 1]
```diff
@@ -543,12 +559,51 @@ Resources:
...
 
+  #############################
+  # SPIKE - New consumer queue
+  #############################
+
+  MockAuditEventQueue:
+    Type: AWS::SQS::Queue
+    Condition: CreateMockTxmaResources
+    Properties:
+      MessageRetentionPeriod: 604800 # 7 days
```
Comment (jkunle) [inline infrastructure/lambda/template.yaml:577]: Do we need 7 days? should just be long enough for the testing need

File: infrastructure/lambda/template.yaml [hunk 2]
```diff
@@ -543,12 +559,51 @@ Resources:
...
+      Enabled: False
       EventSourceArn: !Sub
         - "arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:${queueName}"
-        - queueName: !ImportValue AuditEventQueueName
+        - queueName:
+            Fn::ImportValue: !Sub ${TxmaStackName}-AuditEventQueueName
       FunctionName: !Ref AuditEventConsumerFunction
 
+  #############################
+  # SPIKE - New consumer queue
```
Comment (jkunle) [inline infrastructure/lambda/template.yaml:570]: Can these now be named with a suitable name, it's no longer SPIKE

File: infrastructure/lambda/template.yaml [hunk 3]
```diff
@@ -543,12 +559,51 @@ Resources:
     Condition: IsNotProdLikeEnvironment
     Properties:
       BatchSize: 10
-      Enabled: true
+      Enabled: False
```
Comment (jkunle) [inline infrastructure/lambda/template.yaml:562]: I think this is fine, I have a feeling we can remove this function entirely.

Comment (kiejgordon) [reply infrastructure/lambda/template.yaml:562]: So shall I take this enabled: False out of the PR?

**Review by jkunle — APPROVED** [review]

**Pull request merged**